pkg/internal/sdk/template/brc29/brc29_address_test.go

```diff
@@ -0,0 +1,152 @@
+
+		assert.Error(t, err)
+		require.Nil(t, address)
+	})
+
+	t.Run("return error when nil is passed as sender private key", func(t *testing.T) {
+		var priv *ec.PrivateKey
+
+		address, err := brc29.Address(priv, keyID, recipientPublicKeyHex)
+
+		assert.Error(t, err)
+		require.Nil(t, address)
+	})
+
+	t.Run("return error when nil is passed as recipient public key deriver", func(t *testing.T) {
+		var keyDeriver *sdk.KeyDeriver
+
+		address, err := brc29.Address(senderPrivateKeyHex, keyID, keyDeriver)
+
+		assert.Error(t, err)
+		require.Nil(t, address)
+	})
+
+	t.Run("return error when nil is passed as recipient public key", func(t *testing.T) {
+		var pub *ec.PublicKey
+
+		address, err := brc29.Address(senderPrivateKeyHex, keyID, pub)
+
+		assert.Error(t, err)
+		require.Nil(t, address)
+	})
+
+	t.Run("return valid address created with brc28 with hex string as sender private key source", func(t *testing.T) {
+		address, err := brc29.Address(senderPrivateKeyHex, keyID, recipientPublicKeyHex)
+
+		assert.NoError(t, err)
+		require.NotNil(t, address)
+		require.Equal(t, expectedAddress, address.AddressString)
+	})
+
+	t.Run("return valid address created with brc28 with wif as sender private key source", func(t *testing.T) {
+		address, err := brc29.Address(brc29.WIF(senderWIFString), keyID, recipientPublicKeyHex)
+
+		assert.NoError(t, err)
+		require.NotNil(t, address)
+		require.Equal(t, expectedAddress, address.AddressString)
+	})
+
+	t.Run("return valid address created with brc28 with ec.PrivateKey as sender private key source", func(t *testing.T) {
+		priv, err := ec.PrivateKeyFromHex(senderPrivateKeyHex)
+		require.NoError(t, err)
+
+		address, err := brc29.Address(priv, keyID, recipientPublicKeyHex)
+
+		assert.NoError(t, err)
+		require.NotNil(t, address)
+		require.Equal(t, expectedAddress, address.AddressString)
+	})
+
+	t.Run("return valid address created with brc28 with key deriver as sender private key source", func(t *testing.T) {
+		priv, err := ec.PrivateKeyFromHex(senderPrivateKeyHex)
+		require.NoError(t, err)
+
+		keyDeriver := sdk.NewKeyDeriver(priv)
+
+		address, err := brc29.Address(keyDeriver, keyID, recipientPublicKeyHex)
+
+		assert.NoError(t, err)
+		require.NotNil(t, address)
+		require.Equal(t, expectedAddress, address.AddressString)
+	})
+
+	t.Run("return valid address created with brc28 with ec.PublicKey as receiver public key source", func(t *testing.T) {
+		pub, err := ec.PublicKeyFromString(recipientPublicKeyHex)
+		require.NoError(t, err)
+
+		address, err := brc29.Address(senderPrivateKeyHex, keyID, pub)
+
+		assert.NoError(t, err)
+		require.NotNil(t, address)
+		require.Equal(t, expectedAddress, address.AddressString)
+	})
+
+	t.Run("return testnet address created with brc29", func(t *testing.T) {
+		pub, err := ec.PublicKeyFromString(recipientPublicKeyHex)
+		require.NoError(t, err)
+
+		address, err := brc29.Address(senderPrivateKeyHex, keyID, pub, brc29.WithTestNet())
+
+		assert.NoError(t, err)
```

Comment:
```suggestion
		require.NoError(t, err)
```
Don't think there is need to check next condition if the assertion on error fails here ... therefor i would kindly suggest to use `require`

Reply:
I'm trying to use assert and require such a way, that if the test fail, I have full picture of what was not right,
therefore, even if there is an error, I would like to see if the result was nil.
This way, when sticked to this approach One can spot a weird behaviour, that even if the error was returned, the result wasn't nil and for example even pass all the rest of requirements.